robot/client/client.go

```diff
@@ -251,7 +252,8 @@ func (rc *RobotClient) handleStreamDisconnect(
 
 // New constructs a new RobotClient that is served at the given address. The given
 // context can be used to cancel the operation.
-func New(ctx context.Context, address string, logger golog.Logger, opts ...RobotClientOption) (*RobotClient, error) {
+func New(ctx context.Context, address string, clientLogger *zap.SugaredLogger, opts ...RobotClientOption) (*RobotClient, error) {
```

Comment:
Per my other comment about `AsZap`-confusion, why can't `clientLogger` be of type `logging.ZapCompatibleLogger` here? Then `client.New` could take `zLogger`'s and `golog.Logger`s?

Reply:
Looking into this. I was expecting this to be a `ZapCompatibleLogger`. Hopefully that does work and you just us a needle in a haystack type of bug that I did not want to write.